algorithms.py

```diff
@@ -254,19 +342,20 @@ def __init__(
         self.m = num_loci
         self.r = recombination_rate
         self.migration_matrix = migration_matrix
+        self.num_labels = num_labels
         self.max_segments = max_segments
         self.segment_stack = []
         self.segments = [None for j in range(self.max_segments + 1)]
         for j in range(self.max_segments):
             s = Segment(j + 1)
             self.segments[j + 1] = s
             self.segment_stack.append(s)
-        self.P = [Population(id_) for id_ in range(N)]
-        self.L = FenwickTree(self.max_segments)
+        self.P = [[Population(id_) for id_ in range(N)] for x in range(num_labels)]
```

Comment:
Ah, I've just realised what some of my confusion is. In the C code I've put population as the first index and label as the second. So, each population contains a bunch of labels. 

Do you mind if I change the code here to follow the same convention? It'll be very confusing if they're different.

Reply:
Oh yes totally fine to change which order of the indexes!

Reply:
OK, I'll do the old swaparoony now. Shouldn't take long.